ecs-agent/api/ecs/client/ecs_client_test.go

```diff
@@ -1612,6 +1612,74 @@ func TestFIPSEndpointStateOnFIPSDisabledHosts(t *testing.T) {
 		client.(*ecsClient).standardClient.(*ecsservice.Client).Options().EndpointOptions.UseFIPSEndpoint)
 }
 
+func TestDualStackEndpointStateWhenEndpointGiven(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// Endpoint is given during the call to newMockConfigAccessor function.
+	cfgAccessor := newMockConfigAccessor(ctrl, nil)
+	assert.NotEmpty(t, cfgAccessor.APIEndpoint())
+
+	client, err := NewECSClient(aws.NewCredentialsCache(aws.AnonymousCredentials{}), cfgAccessor, ec2.NewBlackholeEC2MetadataClient(),
+		agentVer)
+	assert.NoError(t, err)
+	assert.Equal(t, aws.DualStackEndpointStateUnset,
+		client.(*ecsClient).standardClient.(*ecsservice.Client).Options().EndpointOptions.UseDualStackEndpoint)
+}
+
+func TestDualStackEndpointStateEnabled(t *testing.T) {
```

Comment:
You may want to use table-driven testing pattern to avoid creating a function per test case. 

Example - https://github.com/aws/amazon-ecs-agent/blob/5b10082318d732e6fae6b84e3c1b876327a6006c/ecs-agent/api/ecs/client/ecs_client_test.go#L309-L439